### PR TITLE
CerebNet support in FastSurfer

### LINKED ISSUE #204: Integrate CerebNet into FastSurfer
## Feature Description
Integrate the CerebNet code, models and scripts into the FastSurfer pipeline
1. Identify redundant code
2. Integrate new code
3. Update call interface (commandline support to call cerebnet through the main entrypoint script (run_fastsurfer.sh)
4. Add validation testing to confirm results are consistent with the original publication

## Motivation
CerebNet provides subsegmentation of the cerebellum which is complementary to FastSurfer. We chose FastSurfer as the main reproducibility, code and support platform.

## Additional Context
Faber J*, Kügler D*, Bahrami E*, Heinz LS, Timmann D, Ernst TM, Deike-Hofmann K, Klockgether T, van de Warrenburg B, van Gaalen J, Reetz K, Romanzetti S, Oz G, Joers JM, Diedrichsen J, ESMMR Study Group, Reuter M. (*co-first). CerebNet: A fast and reliable deep-learning pipeline for detailed cerebellum sub-segmentation. NeuroImage. 2022.
DOI: [ 10.1016/j.neuroimage.2022.119703](http://dx.doi.org/10.1016/j.neuroimage.2022.119703)

### PR DESCRIPTION
**Pull request by dkuegler**
This commit is a current version of the CerebNet integration into FastSurfer.

Unfortunately, I am still working on a couple of issues, for example with the integration changes to the code base, results are not identical between our development and this FastSurfer version.

Furthermore, we have not attached the weights here yet and not included the training and evaluation scripts either.

Several adaptations to FastSurfer that were driven by this implementation have already been merged in several previous PRs.
#243 , #255 , #249 , #239 , #225 , #208

### CODE REVIEW

**Comment by dkuegler** [comment]
This code can be considered to be a functioning beta-version for integration in FastSurfer. Generally, this CerebNet version works for different image resolutions (by downsampling to 1mm) and creates good segmentations for all cases.

This means the code is being moved into a testing phase to ensure the quality and usability is good for full publication/integration.

The CerebNet module also still needs a bit of documentation.

**Comment by fcalcagnoS** [comment]
Would if be possible to make the branch feature/cerebnet public ? Thanks in advance

**Comment by dkuegler** [comment]
Updated the model weights on b2share. Users already using the code here should recreate docker images and delete the checkpoints in Fastsurfer/checkpoints

File: run_fastsurfer.sh [hunk 1]
```diff
@@ -122,6 +127,39 @@ FLAGS:
...
+
+  CEREBELLUM MODULE:
+  --no_cereb              Skip the cerebellum segmentation (CerebNet segmentation)
+  --aparc_aseg_segfile <seg_input>
+                          Name of the segmentation file (similar to aparc+aseg)
+                            for cerebellum localization (typically the output of the
+                            APARC module (see above). Requires an ABSOLUTE Path!
+                            Default location:
+                            \$SUBJECTS_DIR/\$sid/mri/aparc.DKTatlas+aseg.deep.mgz
+	--cereb_segfile <seg_input>
```
Comment (m-reuter) [inline run_fastsurfer.sh:158]: white spaces (probably some hidden tabs here?)

**Comment by XuZ-nins** [comment]
Hi Dr. Kügler, I have not been able to find where the CerebNet module is located (not in the main branch of FastSurfer at least). Is there a way to access it as of now? Thanks!

**Comment by m-reuter** [comment]
This PR was merged into dev on March 2. Feel free to test it by checking out the dev version. Also our release tests are done and look good so far. We expect to release a new version of Fastsurfer that includes cerebnet within 10 days unless anything comes up that requires further investigation.

**Comment by XuZ-nins** [comment]
Hi Dr.  Reuter, thanks for the update! May I also quickly ask if CerebNet is able to segment at original resolution (e.g., 0.4 mm), or only works on the conformed image (1 mm)?

**Comment by m-reuter** [comment]
The current version works only at 1mm. In the future it will be trained similar to VINN also for sub-mm resolutions. 0.4 however is even too small for VINN at the current stage. In order to support resolutions that high, we would need high quality training data for those images, which we don't have (similar for the full brain VINN, which goes down to maybe around .6 or .7 mm isotropic). And if your data is 7T, that would have its additional challenges.

**Pull request merged**